recipes/gdal/pre_3.5.0/test_package/conanfile.py

```diff
@@ -22,7 +22,8 @@ def build(self):
 
     def test(self):
         if can_run(self):
-            if self.options["gdal"].tools:
+            gdal_options = self.options["gdal"] if conan_version < "2" else self.dependencies["gdal"].options
```

Comment:
`self.dependencies` is also valid in v1 :) 
Any special reason that you needed to do this this way?

```suggestion
            gdal_options = self.dependencies["gdal"].options
```

Reply:
for some reason, this is failing depending on conan v1 or conan v2 (with a strange error in the first case).

with the proposed patch conan v2 works but:
```
gdal % conan1 -v
Conan version 1.60.2
gdal % conan1 create pre_3.5.0/conanfile.py 3.4.3@ --build=missing -o '*/*:shared=True'
...
gdal/3.4.3 (test package): Running test()
ERROR: gdal/3.4.3 (test package): Error in test() method, line 25
	gdal_options = self.dependencies["gdal"].options
	AttributeError: 'NoneType' object has no attribute 'dependencies'
```

if I revert this commit:
```
gdal % conan -v
Conan version 2.0.9
conan create pre_3.5.0/conanfile.py --version 3.4.3 --build=missing -o '*/*:shared=True'
...
gdal/3.4.3 (test package): Running test()
ERROR: gdal/3.4.3 (test package): Error in test() method, line 25
	if self.options["gdal"].tools:
	ConanException: option 'tools' doesn't exist
Possible options are []
```